routes/api/upgrade/connect.js

```diff
@@ -0,0 +1,70 @@
+var express = require('express'),
+    router = express.Router();
+
+var responseOk = {
+        'database_setup': {
+            'success': true
+        },
+        'database_migration': {
+            'success': true
+        },
+        'schema_migration': {
+            'success': true
+        },
+        'crowbar_init': {
+            'success': true
+        }
+    },
+    responseFailCrowbar = {
+        'database_setup': {
+            'success': true
+        },
+        'database_migration': {
+            'success': true
+        },
+        'schema_migration': {
+            'success': true
+        },
+        'crowbar_init': {
+            'success': false,
+            'body': {
+                'error': 'crowbar_init: Failed to stop crowbar-init.service'
+            }
+        }
+    },
+    responseFailDatabase = {
+        'database_setup': {
+            'success': false,
+            'body': {
+                'error': 'Could not connect to database server'
+            }
+        },
+        'database_migration': {
+            'success': true
+        },
+        'schema_migration': {
+            'success': true
+        },
+        'crowbar_init': {
+            'success': true
```

Comment:
If the database setup step failed, shouldn't the database migration, schema migration and crowbar init tasks also not be successful?